resources/function_help/json/set_timezone

```diff
@@ -0,0 +1,19 @@
+{
+  "name": "set_timezone",
+  "type": "function",
+  "groups": ["Conversions", "Date and Time"],
+  "description": "Sets the timezone object associated with a datetime value, without changing the date or time components. This function can be used to replace the timezone for a datetime.",
+  "arguments": [{
+    "arg": "datetime",
+    "description": "datetime value"
+  },
+  {
+    "arg": "timezone",
+    "description": "new timezone for datetime"
+  }],
+  "examples": [{
+    "expression": "set_timezone(make_datetime(2020,1,1,10,0,0), timezone_from_id('Australia/Darwin'))",
```

Comment:
Not blocking this PR, but we should update the make_datetime function help/documentation to explicitly state that the created datetime object will be in the machine running QGIS' local timezone.